// test713/routes/mariadb_conn.js
// mariaDB에 connection을 위한 host, post, username, pw 등 기입 후 connection.connect()로 DB 연결

const config = require('../views/info.js');

var mariadb = require('mariadb/callback'); // mariaDB 사용시 반드시 mariadb/callback을 사용한다.

module.exports = function(){
	return{
		init: function(){
			return mariadb.createConnection({
				 host: config.host,
				 port: config.port,
				 user: config.user,
				 password: config.password,
				 database: config.database,
			})
		},
		
		test_open: function(con){
			con.connect(function (err){
				if (err){
					console.error('mariadb connection error');
				}
				else{
					console.info('mariadb is connected successfully.');
				}
			})
		},
		
		test_close: function(con){
			con.end();
		}
	}
};

//  
//async function getUserList(){
//	let conn, rows;
//	try{
//		conn = await pool.getConnection();
//		conn.query('USE test'); // 사용할 DB 명시
//		rows = await conn.query('select * from user');
//	}
//	catch(err){
//		throw err;
//	}
//	finally{
//	if (conn) conn.end();
//	return rows;
//  }
//}
//
//module.exports = { getUserList, }

//return {
//    getConnection: function (callback) {
//      pool.getConnection(callback);
//    },
//    end: function(callback){
//      pool.end(callback);
//    }
//  }
